### PR TITLE
fix: Check inputs validations

### LINKED ISSUE #152: Add validations to form fields
Servers:
- Server URL (not empty, start with protocol, has url shape)

Users:
- email (not empty, email shape)
- username (not empty, max-length of 40, should contain only alphanumeric, dash ('-'), underscore ('_') and dot ('.') characters)
- password (not empty, cannot be less than 6 characters, at least one digit, at least one special character (punctuation, brackets, quotes, etc.), at least one lowercase character, at least one uppercase character)

Project:
- name (not empty, max-length 100)
- description (not empty, max-length 2000)
- repository name (not empty, check gitea for other limitations, max-length 100)

### PR DESCRIPTION
**Pull request by marco-gattelli**
close #152

### CODE REVIEW

File: app/ui/src/Pages/NewProject/pages/Information/Information.tsx [hunk 1]
```diff
@@ -63,9 +67,10 @@ function Information({ showErrors }: Props) {
         }}
         onBlur={() => {
           updateInternalRepositoryValue('slug', generateSlug(name));
+          const isValidName = validateProjectName(name);
           updateError(
             'name',
-            name.length === 0 ? 'This field is mandatory, please fill it.' : ''
+            isValidName === true ? '' : (isValidName as string)
```
Comment (mgs95) [inline app/ui/src/Pages/NewProject/pages/Information/Information.tsx:73]:
```suggestion
            isValidName === true ? '' : `${isValidName}`)
```

Comment (mgs95) [reply app/ui/src/Pages/NewProject/pages/Information/Information.tsx:73]: I am think about this, this is an ugly pattern due to how validations work. Maybe we can create a function that wraps this logic:
```ts
function getErrorMsg(validation: string | boolean): string {
  return validation === true ? '' : (validation as string)
}
```

File: app/ui/src/Pages/NewProject/pages/Information/InformationUtils.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    CHECK.isFieldNotEmpty(name),
+    CHECK.isLengthAllowed(name, 100),
+  ]);
+}
+export function validateProjectDescription(description: string) {
+  return CHECK.getValidationError([
+    CHECK.isFieldNotEmpty(description),
+    CHECK.isLengthAllowed(description, 2000),
+  ]);
+}
```
Comment (mgs95) [inline app/ui/src/Pages/NewProject/pages/Information/InformationUtils.ts:14]: Nice!

File: app/ui/src/Pages/NewProject/pages/RepositoryDetails/components/InternalRepository/InternalRepository.tsx [hunk 3]
```diff
@@ -17,6 +17,11 @@ function validateProjectSlug(value: string): string {
     CHECK.isLowerCase(value),
     CHECK.matches(value, /^[a-z]/, 'Name must start with a lowercase letter'),
     CHECK.matches(value, /.{3,}/, 'Name must contain at least 3 characters'),
+    CHECK.matches(
+      value,
+      /^.{0,100}$/,
+      'Name must contain at most 100 characters'
+    ),
```
Comment (mgs95) [inline app/ui/src/Pages/NewProject/pages/RepositoryDetails/components/InternalRepository/InternalRepository.tsx:24]:
```suggestion
    CHECK.isLengthAllowed(value, 100),
```

File: app/ui/src/Pages/NewUser/NewUserUtils.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+export function verifyPassword(value: string) {
+  return CHECK.getValidationError([
+    CHECK.isFieldNotEmpty(value),
+    CHECK.matches(
+      value,
+      new RegExp(
+        /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{6,}$/g
+      ),
+      'The password length cannot be less than 6 characters, should contain at least one digit, one special character (@$!%*?&), one lowercase character and one uppercase character.'
+    ),
```
Comment (mgs95) [inline app/ui/src/Pages/NewUser/NewUserUtils.ts:24]: We should split this validation.
```suggestion
    CHECK.matches(value, /.{6,}/, 'Password must contain at least 6 characters'),
    CHECK.matches(
      value,
      new RegExp(
        /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]$/g
      ),
      'The password must contain at least one digit, one special character (@$!%*?&), one lowercase character and one uppercase character.'
    ),
```

File: app/ui/src/Pages/NewProject/pages/Information/Information.tsx (hunk 1 above)
Comment (marco-gattelli) [reply app/ui/src/Pages/NewProject/pages/Information/Information.tsx:73]: I like it more the second option, I'm going for it!

**Pull request merged**